x-pack/plugins/ingest_manager/server/errors.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class IngestManagerError {
+  private type: IngestManagerErrorType;
+  private message: string;
+
+  constructor(type: IngestManagerErrorType, message: string) {
+    this.type = type;
+    this.message = message;
+  }
+
+  public getType = (): IngestManagerErrorType => {
```

Comment:
What do you think instead of relaying on type having all of our errors extending IngestManagerError

```
class RegistryError extends IngestManagerError {
}
```

Reply:
See https://github.com/elastic/kibana/issues/66688#issuecomment-633580821

We can do that. We don't use that pattern (inheritance + `instanceof`) at the moment in ingest manager code, and I didn't want to introduce it when it isn't needed.

Reply:
I think it's a really good pattern for `Error` (inheritance + instanceof) and we can keep the getStatusCode method so this stay an internal to this error module, curious to have other thoughts here, maybe @jfsiii? 

But it's really personal preference for me.

Reply:
sounds good to me